packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -910,6 +920,11 @@ export class AppSyncModelVisitor<
             } else {
               addFieldToModel(connectionInfo.connectedModel, connectionInfo.associatedWith);
             }
+            // Add the key to the connected model if it's not explicitly defined
+            //   (either via @index or @belongsTo)
+            if (shouldImputeKeyForUniDirectionalHasMany && hasManyHasImplicitKey(model, connectionInfo)) {
```

Comment:
Since this is within a for loop, we can take advantage of the iterator `field` and pass it into the function. The `field` here should already has `hasMany` attached.

Reply:
Yeah, that's a good point, can simplify the check a bit, and resolve the false positive issue you brought up below. I'll update.